pkg/apis/core/validation/validation.go

```diff
@@ -3167,9 +3167,16 @@ func validateSleepAction(sleep *core.SleepAction, gracePeriod *int64, fldPath *f
 	// We allow gracePeriod to be nil here because the pod in which this SleepAction
 	// is defined might have an invalid grace period defined, and we don't want to
 	// flag another error here when the real problem will already be flagged.
-	if gracePeriod != nil && sleep.Seconds <= 0 || sleep.Seconds > *gracePeriod {
-		invalidStr := fmt.Sprintf("must be greater than 0 and less than terminationGracePeriodSeconds (%d)", *gracePeriod)
-		allErrors = append(allErrors, field.Invalid(fldPath, sleep.Seconds, invalidStr))
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodLifecycleSleepActionAllowZero) {
```

Comment:
We aim to not have feature gates in the validation logic.

You could use the feature gate in the PodValidationOpts.